exporter/jaeger/test/opentelemetry/exporters/jaeger/collector_exporter_test.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+require 'test_helper'
+
+describe OpenTelemetry::Exporter::Jaeger::CollectorExporter do
+  describe '#initialize' do
+    it 'initializes' do
+      exporter = OpenTelemetry::Exporter::Jaeger::CollectorExporter.new
+      _(exporter).wont_be_nil
+    end
+  end
+
+  describe '#export' do
+    let(:exporter) { OpenTelemetry::Exporter::Jaeger::CollectorExporter.new }
+
+    before do
+      OpenTelemetry.tracer_provider = OpenTelemetry::SDK::Trace::TracerProvider.new
+    end
+
+    it 'integrates with collector' do
+      skip unless ENV['TRACING_INTEGRATION_TEST']
+      WebMock.disable_net_connect!(allow: 'localhost')
+      span_data = create_span_data
+      result = exporter.export([span_data])
+      _(result).must_equal(OpenTelemetry::SDK::Trace::Export::SUCCESS)
+    end
```

Comment:
This test fails.

I've run this locally with:
```sh
TRACING_INTEGRATION_TEST=1 bundle exec rake test
```
and the collector:
```sh
./otelcol --config logging.yml
```
where `logging.yml` is:
```yaml
receivers:
  jaeger:
    protocols:
      thrift_compact:
      thrift_http:
  otlp:
    protocols:
      http:
exporters:
  logging:
extensions:
  zpages:
service:
  extensions: [zpages]
  pipelines:
    traces:
      receivers: [otlp, jaeger]
      exporters: [logging]
```

It produces the error:
```
  1) Error:
OpenTelemetry::Exporter::Jaeger::CollectorExporter::#export#test_0001_integrates with collector:
Thrift::ProtocolException: No version identifier, old protocol client?
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/vendor/bundle/ruby/2.7.0/gems/thrift-0.11.0.0/lib/thrift/protocol/binary_protocol.rb:131:in `read_message_begin'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/vendor/bundle/ruby/2.7.0/gems/thrift-0.11.0.0/lib/thrift/client.rb:54:in `receive_message'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/thrift/gen-rb/collector.rb:28:in `recv_submitBatches'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/thrift/gen-rb/collector.rb:20:in `submitBatches'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb:39:in `export'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/test/opentelemetry/exporters/jaeger/collector_exporter_test.rb:27:in `block (3 levels) in <top (required)>'
```

Debugging this will be "fun" 😞

Reply:
> But I am a little confused on how that's ending up in collector logs?

The collector *does* report success:
```
2020-09-09T12:13:39.041-0400	INFO	loggingexporter/logging_exporter.go:296	TraceExporter	{"#spans": 1}
```

But with `strict_read=false` I see the following error from the test:
```
  1) Error:
OpenTelemetry::Exporter::Jaeger::CollectorExporter::#export#test_0001_integrates with collector:
TypeError: no implicit conversion of nil into String
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/vendor/bundle/ruby/2.7.0/gems/thrift-0.11.0.0/lib/thrift/transport/base_transport.rb:87:in `read_all'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/vendor/bundle/ruby/2.7.0/gems/thrift-0.11.0.0/lib/thrift/protocol/binary_protocol.rb:133:in `read_message_begin'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/vendor/bundle/ruby/2.7.0/gems/thrift-0.11.0.0/lib/thrift/client.rb:54:in `receive_message'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/thrift/gen-rb/collector.rb:28:in `recv_submitBatches'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/thrift/gen-rb/collector.rb:20:in `submitBatches'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb:49:in `block in export'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb:75:in `block in untraced'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/api/lib/opentelemetry/trace/tracer.rb:75:in `block in with_span'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/api/lib/opentelemetry/context.rb:62:in `with_value'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/api/lib/opentelemetry/trace/tracer.rb:75:in `with_span'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb:75:in `untraced'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/lib/opentelemetry/exporter/jaeger/collector_exporter.rb:48:in `export'
    /Users/bogsanyi/src/github.com/open-telemetry/opentelemetry-ruby/exporter/jaeger/test/opentelemetry/exporters/jaeger/collector_exporter_test.rb:99:in `block (3 levels) in <top (required)>'
```

Reply:
It's a _new_ error at least 😄 . Hmm, does it need to be coupled with `strict_write=false` as well? We appear to be missing the `version` that's causing issues here? I am very much guessing at this point.